[ZT] mobile WARP client migration (#8579)
cloudflare/cloudflare-docs: content/cloudflare-one/connections/connect-devices/warp/deployment/cloudflare-one-agent-migration.md

## macOS, Windows, and Linux

No action is required for desktop clients at this time. The existing Cloudflare WARP client will continue to support both Zero Trust and 1.1.1.1 functionality.

## iOS and Android

Zero Trust users must migrate from the 1.1.1.1 app to the Cloudflare One Agent app by 2023-12-31.

Organizations can migrate their teams with minimal disruption in one of two modes: [manually](#migrate-manual-deployments) or via a [managed endpoint solution](#migrate-managed-deployments).

### Migration impact

- New Zero Trust features will only be released to the Cloudflare One Agent.
- The 1.1.1.1 app will continue to support the current feature set (and any security updates) through 2023-12-31.
- All Zero Trust functionality will be removed from 1.1.1.1 app versions released after 2023-12-31. Zero Trust will continue to work on 1.1.1.1 app versions released prior to 2023-12-31.

### Migrate manual deployments

If you downloaded and installed the 1.1.1.1 app manually, here are the recommended migration steps:

{{<tabs labels="Android | iOS">}}
{{<tab label="android" no-code="true">}}

> 1. Update the **1.1.1.1** app to version 6.29 or above. The update ensures that 1.1.1.1 can [co-exist](#what-to-do-with-the-old-app) with the new Cloudflare One Agent app.

Reviewer: Note that iOS min version we need people to update to is 6.22. Android is still 6.29

2. If you have enabled [TLS decryption](/cloudflare-one/policies/gateway/http-policies/tls-decryption/), ensure that you have a [Do Not Inspect policy](/cloudflare-one/policies/gateway/initial-setup/http/#bypass-inspection-for-incompatible-applications) in place for the following applications:
    - _Google Services (Do Not Inspect)_
    - _Google Play Store (Do Not Inspect)_
    - _Google (Do Not Inspect)_
    - _Google Drive (Do Not Inspect)_
    - _Google Chat (Do Not Inspect)_
    - _Google Meet (Do Not Inspect)_

This prevents certificate pinning issues when performing the Android migration.

3. [Download](/cloudflare-one/connections/connect-devices/warp/download-warp/#ios) the **Cloudflare One Agent** app.
4. Install the Cloudflare One Agent: 
    {{<render file="_enroll-ios-android.md">}}
5. In the Cloudflare One Agent, turn on the switch to **Connected**.

{{</tab>}}
{{<tab label="ios" no-code="true">}}

> 1. Update the **1.1.1.1** app to version 6.29 or above. The update ensures that 1.1.1.1 can [co-exist](#what-to-do-with-the-old-app) with the new Cloudflare One Agent app.

Author: ```suggestion
1. Update the **1.1.1.1** app to version 6.22 or above. The update ensures that 1.1.1.1 can [co-exist](#what-to-do-with-the-old-app) with the new Cloudflare One Agent app.
```

2. Add the **Cloudflare One Agent** app from the Google Play store. Its application ID is `com.cloudflare.cloudflareoneagent`.

3. Copy your [MDM deployment parameters](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/#android) for the **1.1.1.1** app.

4. Paste the MDM deployment parameters into the **Cloudflare One Agent** app configuration. Make sure that you do not accidentally overwrite the application ID (`com.cloudflare.cloudflareoneagent`).

5. Using your MDM tool, install the **Cloudflare One Agent** on your devices.

6. On Android, the user will need to [re-authenticate](/cloudflare-one/connections/connect-devices/warp/deployment/manual-deployment/#ios-android-and-chromeos) to the new application, following the same onboarding steps they went through initially.

{{</tab>}}
{{<tab label="ios" no-code="true">}}

1. Using your MDM tool, update the **1.1.1.1** app to version 6.29 or above. The update ensures that 1.1.1.1 can co-exist with the new Cloudflare One Agent app during the migration.

> 2. Add the **Cloudflare One Agent** app from the App store. Its application ID is `com.cloudflare.cloudflareoneagent`.

Author: Is the app ID the same for Android and iOS?

Reviewer 2: Yes app ID is same for Android & iOS.

3. Copy your [MDM deployment parameters](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/#android) for the **1.1.1.1** app.

> 4. Paste the MDM deployment parameters into the **Cloudflare One Agent** app configuration. Make sure that you do not accidentally overwrite the application ID (`com.cloudflare.cloudflareoneagent`).

Author: same question regarding the iOS application ID

Reviewer 2: Yes app ID is same for Android & iOS, that is: com.cloudflare.cloudflareoneagent

5. Using your MDM tool, install the **Cloudflare One Agent** on your devices.

On iOS, the user does not need to re-authenticate — registration data from the 1.1.1.1 app is automatically migrated to the new Cloudflare One Agent app.

{{</tab>}}
{{</tabs>}}

> Once users have enrolled, the migration process is complete.  The 1.1.1.1 app will revert to [consumer mode](#what-to-do-with-the-old-app) and ignore the existing MDM configuration profile. If you do not wish to keep the 1.1.1.1 app, you may uninstall it and delete its MDM configuration.

Reviewer 3: Super tiny nit, but there's an extra space after "complete."

---
pcx_content_type: how-to
title: Migrate to the Cloudflare One Agent
weight: 12
---

# Migrate to the Cloudflare One Agent

> Cloudflare has started splitting the [Zero Trust WARP client](/cloudflare-one/connections/connect-devices/warp/) and the [consumer WARP client](/warp-client/) into two separate applications. Users who are logged in to a Zero Trust organization on the WARP client will need to migrate to the new Cloudflare One Agent. Users who are running the WARP client in consumer / 1.1.1.1 mode are not impacted by this change.

Reviewer: Two comments for this line:

1. One of our key points when talking to customers is that our ZT app is built off the shared consumer platform used by millions of devices. With the new Cloudflare One Agent that will still be true, we're just doing a more surgical separation while still sharing a lot of the common core. Need to figure out how to rephrase this paragraph to make sure we don't lose that message.

2. `Users who are logged in to a Zero Trust organization on the WARP client will need to migrate to the new Cloudflare One Agent` should end with a date by when that will need to happen. I believe Sept 30th, 2023 is our public date but should confirm with Shiv.

## macOS, Windows, and Linux

> The Cloudflare One Agent for desktop is coming soon. No action is required at this time.

Reviewer: Would replace "Soon" with something more like "At a later date". We just moved out the work from Q2 to focus on other areas so "Soon" here probably means end of 2023 or beginning of 2024.

## iOS and Android

> Starting on 2023-04-24, new Zero Trust features will be invested into the **Cloudflare One Agent** application. The **1.1.1.1** application will continue to support the current feature set until 2023-09-30.

Reviewer: Can we use full month written dates instead of those odd european dates :) Or is this a standard we do elsewhere? 

Also `features will be invested into` is a weird phrasing. We should probably consider removing the "Starting on" part.

Author: That's the new date format for dev docs and the dash: https://wiki.cfdata.org/display/Prod/Dates+and+Times

### Migrate managed deployments

If you deployed the WARP client with an [MDM provider](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/), perform the migration as follows:

> 1. Uninstall the **1.1.1.1** application.

Reviewer 4: Might be worth explaining whether it's important to uninstall the 1.1.1.1 app before installing the new Cloudflare One app, and explain what happens if both are installed at the same time.

If it's important that 1.1.1.1 be removed before pushing the new app, we need to explain that they should aim to batch the changes, so the new configuration that removes 1.1.1.1 and requires Cloudflare One arrives at the same time on the device.

Otherwise, if they do it in two steps, they might have 1.1.1.1 removed for an undeterminate amount of time until Cloudflare One is installed and active.

We also need to explain what happens if both app are installed at the same time.

Reviewer 4: @naga-cf can you help with verbiage here. I think what we have right now is a good start but a bit too light on the details.

2. Update your MDM configuration with the new application ID:

>     - For Android, the bundle ID is `com.cloudflare.coudflareoneagent`.

Reviewer 4: ```suggestion
    - For Android, the new application ID is `com.cloudflare.cloudflareoneagent`.
```

>     - For iOS, the application ID is `?`.

Author: need iOS ID

>     - For iOS, the application ID is `?`.

Reviewer 4: ```suggestion
    - For iOS, the new application ID is `com.cloudflare.cloudflareoneagent`.
```

>     Your [WARP deployment parameters](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/parameters/) can remain the same.

Reviewer 4: ```suggestion
    The other [WARP deployment parameters](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/parameters/) you have configured remain the same.
```

3. Push the change to install the **Cloudflare One Agent**.

> The user will need to re-authenticate to the new application.

Author: On Android, the user will be prompted to re-authenticate to the new application. 

What's the behavior on iOS?

Reviewer 4: @naga-cf ^

### Migrate manual deployments

If you downloaded and installed the WARP client manually, perform the following steps on your device:

1. [Uninstall](/cloudflare-one/connections/connect-devices/warp/remove-warp/#ios-and-android) the **1.1.1.1** application.
2. [Download](/cloudflare-one/connections/connect-devices/warp/download-warp/) the **Cloudflare One Agent**.
3. [Re-enroll the device](/cloudflare-one/connections/connect-devices/warp/deployment/manual-deployment/#ios-android-and-chromeos) in your Zero Trust organization.
4. Complete the authentication steps required by your organization.

> The migration process is now complete.

Reviewer 4: ```suggestion
Once you have enrolled, the migration process is complete.
```

## macOS, Windows, and Linux

Cloudflare One Agent for desktop is coming at a later date. No action is required at this time.

## iOS and Android

Zero Trust users must migrate from the 1.1.1.1 app to the Cloudflare One Agent app by 2023-09-30.

### Migration impact
- New Zero Trust features will be added to the Cloudflare One Agent.
- 1.1.1.1 will continue to support the current feature set until 2023-09-30.
- After 2023-09-30, all Zero Trust functionality will be removed from 1.1.1.1.

### Migrate managed deployments

If you deployed the WARP client with an [MDM provider](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/), perform the migration as follows:

1. Configure your MDM to uninstall **1.1.1.1**.

>     While both 1.1.1.1 and Cloudflare One Agent can exist on the device, iOS and Android will only allow one of these applications to run at a time.

Reviewer 4: I think this is a good start compared to before, but I think we need to be give more details than this. @naga-cf can you help with the verbiage here?

For reference, here's what I was suggesting before:

> Might be worth explaining whether it's important to uninstall the 1.1.1.1 app before installing the new Cloudflare One app, and explain what happens if both are installed at the same time.
> 
> If it's important that 1.1.1.1 be removed before pushing the new app, we need to explain that they should aim to batch the changes, so the new configuration that removes 1.1.1.1 and requires Cloudflare One arrives at the same time on the device.
> 
> Otherwise, if they do it in two steps, they might have 1.1.1.1 removed for an undeterminate amount of time until Cloudflare One is installed and active.
> 
> We also need to explain what happens if both app are installed at the same time.

## macOS, Windows, and Linux

Cloudflare will release dedicated Cloudflare One agents for Windows, Mac and Linux at a later date. No action is required at this time. The existing Cloudflare WARP client will continue to support both Zero Trust and 1.1.1.1 w/ WARP functionality.

## iOS and Android

Zero Trust users must migrate from the 1.1.1.1 w/ WARP app to the Cloudflare One Agent app by 2023-09-30.

Organizations can migrate their teams with minimal disruption in one of two modes: [manually](#migrate-manual-deployments) or via a [managed endpoint solution](#migrate-managed-deployments).

### Migration impact

- New Zero Trust features will only be released to the Cloudflare One Agent.
- 1.1.1.1 w/ WARP will continue to support the current feature set (and any security updates) through 2023-09-30.
- After 2023-09-30, all Zero Trust functionality will be removed from 1.1.1.1 w/ WARP. Users will no longer be able to log in to your Zero Trust organization through 1.1.1.1 w/ WARP.

### Migrate manual deployments

If you downloaded and installed the WARP client manually, perform the following steps on your device:

> 1. [Uninstall](/cloudflare-one/connections/connect-devices/warp/remove-warp/#ios-and-android) the **1.1.1.1** application.

Reviewer 5: @naga-cf should this section be updated for any auto-migration that will happen if you have both apps installed? If so this section should describe how to make sure that migration happen (something like open the new app while both are installed). Otherwise +1 to moving the uninstall to the end of the instructions

2. [Download](/cloudflare-one/connections/connect-devices/warp/download-warp/) the **Cloudflare One Agent**.

> 3. [Re-enroll the device](/cloudflare-one/connections/connect-devices/warp/deployment/manual-deployment/#ios-android-and-chromeos) in your Zero Trust organization.

Reviewer 5: Since this is specific to iOS & Android it might be useful indicate that this is done through the app:
```suggestion
3. Open the **Cloudflare One Agent** app to [re-enroll the device](/cloudflare-one/connections/connect-devices/warp/deployment/manual-deployment/#ios-android-and-chromeos) in your Zero Trust organization.
```

4. Complete the authentication steps required by your organization.

Once you have enrolled, the migration process is complete.

> 

Reviewer 6: Add section here:

## What do I do with the old 1.1.1.1 app?
You can now use the 1.1.1.1 app for _personal browsing_.

This means that your traffic will be encrypted and privately routed via Cloudflare's network, and your employer or managing organization will not be able to see any of your browsing activity when connected via 1.1.1.1.

If you need to access your company's resources, you must use the Cloudflare One app. If you would not like to use the 1.1.1.1 app for private personal browsing, you may delete it.

### Migrate managed deployments

If you deployed the WARP client with an [MDM provider](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/), perform the migration as follows:

1. Configure your MDM to uninstall the **1.1.1.1** application.

    While both 1.1.1.1 and Cloudflare One Agent can exist on the device, iOS and Android will only allow one of these applications to run at a time.

2. Update your MDM configuration with the new application ID: `com.cloudflare.cloudflareoneagent`

    The other [WARP deployment parameters](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/parameters/) you have configured remain the same.

> 3. Push the change to remove **1.1.1.1** and install **Cloudflare One Agent**.

Reviewer 5: What does `Push the change` mean in this context?

> 4. Users will need to re-authenticate to the new application, following the same onboarding steps they went through initially.

Reviewer 5: Is there a guide for the re-authentication/onboarding that we could link to from here?

The Cloudflare One Agent supports all existing Zero Trust functionality. The underlying connection technology remains the same, and improvements made to performance and reliability based on feedback from 1.1.1.1 w/ WARP users will continue to be built into the Cloudflare One Agent.

## macOS, Windows, and Linux

No action is required for desktop clients at this time. The existing Cloudflare WARP client will continue to support both Zero Trust and 1.1.1.1 w/ WARP functionality.

## iOS and Android

Zero Trust users must migrate from the 1.1.1.1 w/ WARP app to the Cloudflare One Agent app by 2023-09-30.

Organizations can migrate their teams with minimal disruption in one of two modes: [manually](#migrate-manual-deployments) or via a [managed endpoint solution](#migrate-managed-deployments).

### Migration impact

- New Zero Trust features will only be released to the Cloudflare One Agent.
- 1.1.1.1 w/ WARP will continue to support the current feature set (and any security updates) through 2023-09-30.
- After 2023-09-30, all Zero Trust functionality will be removed from 1.1.1.1 w/ WARP. Users will no longer be able to log in to your Zero Trust organization through 1.1.1.1 w/ WARP.

> ### Migrate manual deployments

Reviewer 2: Shruthi Srinath (our QA) and I discussed and think below recommended scenario/steps would help customer for easy migration. Happy to discuss if anyone have any questions.

**Recommended approach (for Manual migration)**
**With 1.1.1.1 already installed and user in ZT mode (1.1.1.1 app is on version 6.28 (android)/ xx (iOS) or below)**
1. Uninstall 1.1.1.1 app
2. Install CFOne app
3. Click "Next" on CFOne app for onboarding screen
4. Click "Accept" on CFOne app for onboarding/privacy screen
5. Enter "organization name" in CFOne app and re-enroll the device in your Zero Trust organization

**With 1.1.1.1 already installed and user in ZT mode with same organization (1.1.1.1 app is on version 6.29 (android)/ xx (iOS) or above)**
1. Update 1.1.1.1 app
2. Install CFOne app
3. Click "Next" on CFOne app for onboarding screen
4. Click "Accept" on CFOne app for onboarding/privacy screen
5. Enter "organization name" in CFOne app and re-enroll the device in your Zero Trust organization (same org as in 1.1.1.1 app)
6. Connect to the service (toggle on) in CFOne app and your 1.1.1.1 app data would reset (user would be logged out of ZT on 1.1.1.1 app if CFOne and 1.1.1.1 app have same organization)

**With 1.1.1.1 app not installed**
1. Install CFOne app
2. Click "Next" on CFOne app for onboarding screen
3. Click "Accept" on CFOne app for onboarding/privacy screen
4. Enter "organization name" in CFOne app and re-enroll the device in your Zero Trust organization

**Other approach**:
**With 1.1.1.1 app already installed and user in ZT mode with different organization (1.1.1.1 app is on version 6.29 (android)/ xx (iOS) or above)**
1. Update 1.1.1.1 app
2. Install CFOne app
3. Click "Next" on CFOne app for onboarding screen
4. Click "Accept" on CFOne app for onboarding/privacy screen
5. Enter "organization name" in CFOne app and re-enroll the device in your Zero Trust organization (different org then in 1.1.1.1 app)
6. Connect to the service (toggle on) in CFOne app and your 1.1.1.1 app data would NOT reset (user is logged into ZT and 1.1.1.1 app into a different organizations)

**With 1.1.1.1 app NOT in ZT mode and CFOne app installed**
1. Clicking Log into ZT button on 1.1.1.1 app would navigate user to CFOne app

**Ref**: 
https://docs.google.com/document/d/1U9azuHFRuBPzKJeLmFBtfU2joZn-HAWW-50OLL_U8Hw/edit
In this reference document, we have scenario 2 which is just for internal reference and we don't want to share and confuse our customers.

Author: Can we provide a single recommended approach? An end user may be overwhelmed by the number of different options here.
@spatel-cf

Reviewer 2: There are 3 scenarios users device could be in (wrt 1.1.1.1 app) when we send out the comms/share Cloudflare docs with our customers. Thats the reason we listed all the 3 scenarios, but its PM's call @spatel-cf any thought?. Also please feel free to modify the wordings as those might not be accurate for publishing the article :)

Reviewer 7: I think we should consolidate the three scenarios: 1) with old 1.1.1.1 2) with newer version of 1.1.1.1 and 3) with no 1.1.1.1 installed on the machine. 

So this way, our docs would state the three different options of what the user will need to do when 1.1.1. is installed (both older/newer versions) and then instructions when there is no 1.1.1.1 installed (new install). Does this make sense?

Author: What are the downsides if you don't uninstall or update 1.1.1.1? Others have suggested that installing CF1 should be step 1: https://github.com/cloudflare/cloudflare-docs/pull/8579#discussion_r1198196948

Author: > So this way, our docs would state the three different options of what the user will need to do when 1.1.1. is installed (both older/newer versions) and then instructions when there is no 1.1.1.1 installed (new install). Does this make sense?

Since this is a migration guide, I think we can skip the new install scenario. Those instructions will be on our manual deployment page.

Reviewer 2: If user does not update 1.1.1.1 app:
User would not land in a consumer mode on 1.1.1.1 app (if already logged into ZT). Both the application would be logged into same or different org. 

**If different org**: it doesn't matter
**If same org**: any dash changes triggers notification on the client and we could not guarantee which application would get notification first. So one of the possible scenario could be that user tunnel is connected for CFOne app but with notification received in 1.1.1.1 app first (where switch-lock=true) then CFOne tunnel would be disconnected automatically causing bad user experience.

Any dash change/notification arriving to client wrt to restarting the existing tunnel would enter into a race condition and could cause customer escalations. So update should be the first priority but if customer has decided that for sure they are going to use CFOne app then just uninstall 1.1.1.1 app.

With 1.1.1.1 update we reset/clear ZT data from 1.1.1.1 app if CFOne app is also logged into the same org (but we cannot do that on older version)

Reviewer 2: > I think we should consolidate the three scenarios: 1) with old 1.1.1.1 2) with newer version of 1.1.1.1 and 3) with no 1.1.1.1 installed on the machine.
> 
> So this way, our docs would state the three different options of what the user will need to do when 1.1.1. is installed (both older/newer versions) and then instructions when there is no 1.1.1.1 installed (new install). Does this make sense?

**With 1.1.1.1 already installed and user in ZT mode (1.1.1.1 app is on version 6.28 (android)/ xx (iOS) or below)**
1. Uninstall 1.1.1.1 app or follow scenario 2 (link to **With 1.1.1.1 already installed and user in ZT mode with same organization (1.1.1.1 app is on version 6.29 (android)/ xx (iOS) or above)**)
2. Follow steps below (link to Follow the common steps below)

**With 1.1.1.1 already installed and user in ZT mode with same organization (1.1.1.1 app is on version 6.29 (android)/ xx (iOS) or above)**
1. Update 1.1.1.1 app
2. Follow steps below (link to Follow the common steps below)
3. Connect to the service (toggle on) in CFOne app and your 1.1.1.1 app data would reset (user would be logged out of ZT on 1.1.1.1 app if CFOne and 1.1.1.1 app have same organization)

**Follow common steps**
1. Install CFOne app
2. Click "Next" on CFOne app for onboarding screen
3. Click "Accept" on CFOne app for onboarding/privacy screen
4. Enter "organization name" in CFOne app and re-enroll the device in your Zero Trust organization

Also I strongly recommend not to exclude other approach section and its good that we share it in Cloudflare docs with our customers. 

cc: @spatel-cf

Author: Ok I made a couple of updates. There is a single recommended workflow, but I think it captures the scenarios listed above.

The Cloudflare One Agent supports all existing Zero Trust functionality. The underlying connection technology remains the same, and improvements made to performance and reliability based on feedback from 1.1.1.1 w/ WARP users will continue to be built into the Cloudflare One Agent.

## macOS, Windows, and Linux

No action is required for desktop clients at this time. The existing Cloudflare WARP client will continue to support both Zero Trust and 1.1.1.1 w/ WARP functionality.

## iOS and Android

Zero Trust users must migrate from the 1.1.1.1 w/ WARP app to the Cloudflare One Agent app by 2023-09-30.

Organizations can migrate their teams with minimal disruption in one of two modes: [manually](#migrate-manual-deployments) or via a [managed endpoint solution](#migrate-managed-deployments).

### Migration impact

> - New Zero Trust features will only be released to the Cloudflare One Agent.

Author: ```suggestion
- Starting in July 2023, new Zero Trust features will only be released to the Cloudflare One Agent.
```

- 1.1.1.1 w/ WARP will continue to support the current feature set (and any security updates) through 2023-09-30.

> - After 2023-09-30, all Zero Trust functionality will be removed from 1.1.1.1 w/ WARP. Users will no longer be able to log in to your Zero Trust organization through 1.1.1.1 w/ WARP.

Reviewer 8: We should probably mention about newer 1.1.1.1 w/WARP versions post 2023-09-30.

After reading through the statement, it sounded to me as if ZT functionality will not work on ALL 1.1.1.1 w/ WARP versions which is not the case since ZT will continue to work as before on any versions prior to 2023-09-30.

### Migrate manual deployments

> If you downloaded and installed the WARP client manually, here are the recommended migration steps:

Reviewer 8: We are referring the current 1.1.1.1 w/ WARP app with multiple names. It was called 1.1.1.1 w/ WARP on line 19 and we are calling it as WARP client here and call it as 1.1.1.1. app in line 33.

To access your company's resources, you must use the Cloudflare One Agent app.

You can use the 1.1.1.1 app for personal browsing. When connected to 1.1.1.1 w/ WARP, your traffic will be encrypted and privately routed via Cloudflare's network, and your employer will not be able to see any of your browsing activity. To learn more about consumer WARP services, refer to [WARP client](/warp-client/).

If you do not wish to use the old 1.1.1.1 app for personal browsing, you may [uninstall](/cloudflare-one/connections/connect-devices/warp/remove-warp/#ios-and-android) it.

### Migrate managed deployments

If you deployed the WARP client with an [MDM provider](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/), perform the migration as follows:

1. Configure your MDM to uninstall the **1.1.1.1** application.

2. Update your MDM configuration with the new application ID: `com.cloudflare.cloudflareoneagent`

    The other [WARP deployment parameters](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/parameters/) you have configured remain the same.

> 3. To minimize user confusion, we recommend simultaneously removing **1.1.1.1** and installing **Cloudflare One Agent**.

Reviewer 4: Not all MDMs allow uninstalling an app and pushing a new one simultaneously. Additionally, there is a risk that 1.1.1.1 would be deleted first, and Cloudflare One Agent installed second. On iOS, my expectation is that this would lead to the registration being deleted.

@naga-cf can you check with Rodrigo and Shruti whether that'd be the case? And if so, what we should recommend instead?

Reviewer 4: For example, in Intune, companies generally assign or remove an app to/from a group.
So I wouldn't be able to simultaneously add CF1 and remove 1.1.1.1. Instead I'd need to add CF1 to the group, then probably wait until all devices have picked up the change, and then wait for a few days before removing 1.1.1.1 to make sure the migration has happened (especially for devices that might have been offline while I was making the change).

Reviewer 9: I think the term simultaneously is not correct here, something after the migration period would be better to remove settings from the Consumer app. Because you don't need to remove the application, but the MDM settings.

4. On Android, the user will need to [re-authenticate](/cloudflare-one/connections/connect-devices/warp/deployment/manual-deployment/#ios-android-and-chromeos) to the new application, following the same onboarding steps they went through initially.

>     On iOS, the user does not need to re-authenticate — registration data from the 1.1.1.1 w/ WARP app is automatically migrated to the new Cloudflare One Agent app.

Reviewer 8: We are asking admins to uninstall 1.1.1.1 w/ WARP app in point (1) above and once the app is deleted, registration data will be removed as well so migration will fail.
